server/src/test/java/org/elasticsearch/search/aggregations/metrics/HyperLogLogPlusPlusTests.java

```diff
@@ -171,4 +171,16 @@ public long addWithoutBreaking(long bytes) {
 
         assertThat(total.get(), equalTo(0L));
     }
+
+    public void testRetrieveCardinality() {
+        final int p = randomIntBetween(MIN_PRECISION, MAX_PRECISION);
+        final HyperLogLogPlusPlus counts = new HyperLogLogPlusPlus(p, BigArrays.NON_RECYCLING_INSTANCE, 1);
+        int bucket = randomInt(100);
+        counts.collect(bucket, -8688952809613614893L);
```

Comment:
Probably worth a comment about why this particular random number. If it is to trip the problem, can we somehow assert that it tripped with this number? I feel like we could easily end up in a position where the test doesn't do anything if we don't have some package private method to check to see if this thing happened *somehow*.

Reply:
The only condition to trigger it is to add one value to one bucket. I change it to a random long.